### DISCUSSION TITLE
Parse Events?

### DISCUSSION THREAD

**Discussion by appleseed-iii** [foundry-rs/foundry#2258] [Q&A]
Is there a way to parse events with `cast`? The `cast` [README](https://github.com/foundry-rs/foundry/blob/master/cast/README.md) appears to reference `cast events`, but the CLI doesn't seem to think that's a valid call.

```sh
$ cast events
error: Found argument 'events' which wasn't expected, or isn't valid in this context

USAGE:
    cast <SUBCOMMAND>
```

**Comment by onbjerg** [comment]
It's marked as not done (checkbox not filled in). If you want to get the signature of an event use `cast 4byte-event <topic 0>`. Can you elaborate a bit more on what you want to do?

**Reply by appleseed-iii** [reply]
yea I'm hoping to query / filter historical events on a contract similar to how I might using ethers https://docs.ethers.io/v5/concepts/events/

A current use case is when we have a contract with non-enumerable roles. I'd like to query the correct Events to determine which addresses have been given a role.

**Reply by onbjerg** [reply]
Ah, right. Yeah, that feature is not implemented yet. It's tracked somewhere in an issue and I think there might have been an attempt at a PR as well.

Edit: here https://github.com/foundry-rs/foundry/pull/869
